tests/phpunit/tests/rest-api/rest-posts-controller.php

```diff
@@ -1527,6 +1528,32 @@ public function test_get_items_not_sticky_with_exclude_no_sticky_posts() {
 		$this->assertPostsWhere( " AND {posts}.ID NOT IN ($id3) AND {posts}.post_type = 'post' AND (({posts}.post_status = 'publish'))" );
 	}
 
+	public function test_get_items_with_custom_search_columns() {
+		$id1 = self::factory()->post->create(
+			array(
+				'post_title'   => 'Title contain foo and bar',
+				'post_content' => 'Content contain bar',
+				'post_excerpt' => 'Excerpt contain baz',
+			)
+		);
+		$id2 = self::factory()->post->create(
+			array(
+				'post_title'   => 'Title contain baz',
+				'post_content' => 'Content contain foo and bar',
+				'post_excerpt' => 'Excerpt contain foo, bar and baz',
+			)
+		);
+
+		$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+		$request->set_param( 'search', 'foo bar' );
+		$request->set_param( 'search_columns', array( 'post_title' ) );
+		$response = rest_get_server()->dispatch( $request );
+		$this->assertSame( 200, $response->get_status() );
+		$data = $response->get_data();
+		$this->assertCount( 1, $data );
+		$this->assertSame( $id1, $data[0]['id'] );
```

Comment:
While I understand that this test class lacks a final `$message` parameter in the assertions of other test methods, I think it would be good to keep consistent with the rest of the PR and add them to this new test method, to save ourselves time in the future. 🙂

Reply:
Added. I'll keep that in mind in the future when writing / updating tests with multiple assertions.

Reply:
Thanks!

Also, I thought I'd pasted the link to the standards reference on this, but I didn't. Oops! For your reference, see [here](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#using-assertions) (the last paragraph).